packages/astro/src/runtime/server/render/component.ts

```diff
@@ -92,7 +92,7 @@ async function renderFrameworkComponent(
 		displayName,
 	};
 
-	const { hydration, isPage, props } = extractDirectives(_props, clientDirectives);
+	const { hydration, isPage, props, propsWithoutTransitionAttributes } = extractDirectives(_props, clientDirectives);
```

Comment:
Oh I actually thought `propsWithoutTransitionAttributes` would replace `props`, but I think this is fine too. The props extraction are now all in one spot 👍